### DISCUSSION TITLE
How to handle complaints like SendGrid, Mailgun, Postmark,AWS...?

### DISCUSSION THREAD

**Discussion by benstaf** [postalserver/postal#3083] [Help with using Postal]
Hi, does Postal have a way to deal with complaints and spam reports, like it has for bounces?

How are complaints dealt with?

Documentation doesn't say anything:
https://docs.postalserver.io/developer/webhooks

Compare with SendGrid:

https://sendgrid.com/en-us/blog/in-praise-of-spam-complaints

Mailgun:

https://help.mailgun.com/hc/en-us/articles/203661564-Events-Logs-Message-Event-Types

Postmark:

https://postmarkapp.com/support/article/815-what-are-bounces-and-spam-complaints

Amazon:
https://aws.amazon.com/blogs/messaging-and-targeting/amazon-ses-set-up-notifications-for-bounces-and-complaints/

**Comment by willpower232** [comment]
Postals main focus is to send transactional messages. Obviously it could also send marketing messages but it doesn't manage your email subscription list, any content or templating, or unsubscribe or spam reports. It does not even want to hold on to messages you have sent for very long.

As per #33 which you have already seen, it is down to you to provide the relevant header on the message if it is something you wish to handle.

**Comment by benstaf** [comment]
Postal is advertised as a "complete and fully featured" open-source alternative to "SendGrid, Mailgun and Postmark", but lacks this basic feature 

![Screenshot_2024-08-08-13-51-50-884_com android chrome-edit](https://github.com/user-attachments/assets/4862543f-cc1c-4914-9fc7-166b1dae7110)


There's probably a workaround indeed, and I will post it as soon as I solve this bad surprise, but I am surprised nobody did it before.

**Reply by charlesmudy** [reply]
@benstaf do you have solution to this?

**Reply by benstaf** [reply]
there's a solution for hard/soft bounces, but I didn't find a solution for complaints yet
